aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -21,127 +21,134 @@
 // If you're writing something more specialized, please don't try to make them
 // work for your case, but just write something new instead.
 
-namespace at {
-namespace native {
+namespace at { namespace native {
 
-Tensor& fill_(Tensor& self, Scalar value) {
-  return self._fill_(value);
+Tensor& _fill__cpu(Tensor& self, Scalar value_) {
+  fillImpl(self, value_);
+  return self;
 }
 
-Tensor& fill_(Tensor& self, const Tensor& value) {
-  return self._fill_(value);
+Tensor& _fill__cpu(Tensor& self, const Tensor& value__) {
+  Tensor value_ = value__;
+  if (value_.dim() == 0)
+    value_ = value_.view(1);
+  Scalar value(value_[0]);
+  fillImpl(self, value);
```

Comment:
Uhhh what's going on in here with all those underscores. Can't you just `at::Scalar(value__)`?

Reply:
No, that conversion uses fill.